frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/vm-settings-tab/provision-source.tsx

```diff
@@ -0,0 +1,95 @@
+import { ProvisionSource } from '../../../../constants/vm/provision-source';
+import { FormFieldRow } from '../../form/form-field-row';
+import { FormField, FormFieldType } from '../../form/form-field';
+import { iGetFieldValue } from '../../selectors/immutable/field';
+import { VMSettingsField } from '../../types';
+import { getPlaceholder } from '../../utils/renderable-field-utils';
+
+export const ProvisionSourceComponent: React.FC<ProvisionSourceComponentProps> = React.memo(
+  ({
+    provisionSourceField,
+    onChange,
+    goToStorageStep,
+    goToNetworkingStep,
+    getProvisionSourceAttribute,
+  }) => {
+    const provisionSourceValue = iGetFieldValue(provisionSourceField);
+    const storageBtn = (
+      <Button
+        isDisabled={!goToStorageStep}
+        isInline
+        onClick={goToStorageStep}
+        variant={ButtonVariant.link}
+      >
+        <strong>Storage</strong>
+      </Button>
+    );
+    const networkBtn = (
+      <Button
+        isDisabled={!goToNetworkingStep}
+        isInline
+        onClick={goToNetworkingStep}
+        variant={ButtonVariant.link}
+      >
+        <strong>Networking</strong>
+      </Button>
+    );
+    const getStorageMsg = () => {
+      switch (provisionSourceValue) {
+        case ProvisionSource.URL.toString():
+          return <>Enter URL here or edit the mounted disk in the {storageBtn} step</>;
+        case ProvisionSource.CONTAINER.toString():
+          return <>Enter container image here or edit the mounted disk in the {storageBtn} step</>;
+        case ProvisionSource.DISK.toString():
+          return <>Add a source disk in the {storageBtn} step</>;
+        default:
+          return null;
+      }
+    };
+
+    const provisionSourceDiskHelpMsg = (
+      <div className="pf-c-form__helper-text" aria-live="polite">
+        {getStorageMsg()}
+      </div>
+    );
+    const provisionSourceNetHelpMsg = (
+      <div className="pf-c-form__helper-text" aria-live="polite">
+        Add a network interface in the {networkBtn} step
+      </div>
+    );
+
+    return (
+      <FormFieldRow field={provisionSourceField} fieldType={FormFieldType.SELECT}>
+        <FormField>
+          <FormSelect onChange={(v) => onChange(VMSettingsField.PROVISION_SOURCE_TYPE, v)}>
+            <FormSelectPlaceholderOption
+              placeholder={getPlaceholder(VMSettingsField.PROVISION_SOURCE_TYPE)}
+              isDisabled={!!provisionSourceValue}
+            />
+            {(getProvisionSourceAttribute('sources') || []).map((source) => (
+              <FormSelectOption key={source} value={source} label={source} />
+            ))}
+          </FormSelect>
+        </FormField>
+        {[
+          ProvisionSource.URL.toString(),
+          ProvisionSource.CONTAINER.toString(),
+          ProvisionSource.DISK.toString(),
+        ].includes(provisionSourceValue) && provisionSourceDiskHelpMsg}
```

Comment:
it would be nice to have each one of these as a small component as well

Reply:
it's a component now :compass:

Reply:
using callbacks :-)